### PR TITLE
Add an RTIC example

### PR DESCRIPTION
**Pull request by phoracek**
This example presents integration with RTIC [1]. Built for STM32F4
Discovery, toggles an onboard LED upon an interrupt fired by an onboard
button.

[1] https://rtic.rs/

Signed-off-by: Petr Horacek <hrck@protonmail.com>

### CODE REVIEW

File: examples/rtic.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        let gpioa = ctx.device.GPIOA.split();
+        let mut button = gpioa.pa0.into_pull_down_input();
+        button.make_interrupt_source(&mut syscfg);
+        button.enable_interrupt(&mut ctx.device.EXTI);
+        button.trigger_on_edge(&mut ctx.device.EXTI, Edge::RISING);
+
+        stm32::NVIC::unpend(stm32::Interrupt::EXTI0);
+        unsafe {
+            stm32::NVIC::unmask(stm32::Interrupt::EXTI0);
+        };
```
Comment (samcrow) [inline examples/rtic.rs:38]: The RTIC-generated code unmasks the interrupt before calling init, so we don't need to do that again here.

Comment (phoracek) [reply examples/rtic.rs]: Oh, I missed that. Thanks for the review. Fixed

**Review by therealprof — APPROVED** [review]
Very nice, thanks!

bors r+

**Review by therealprof — APPROVED** [review]
bors r+

**Pull request merged**